rplugin/python3/deoplete/util.py

```diff
@@ -145,3 +145,26 @@ def strwidth(string):
 def charwidth(c):
     wc = unicodedata.east_asian_width(c)
     return 2 if wc == 'F' or wc == 'W' else 1
+
+
+def itersource(vim, context, sources):
+    filetypes = context['filetype'].split('.')
+    ignore_sources = set()
+    for ft in filetypes:
+        ignores = get_buffer_config(vim, ft, 'b:deoplete_ignore_sources',
+                                    'g:deoplete#ignore_sources', '{}')
```

Comment:
I got inconsistent results from `deoplete#util#get_buffer_config()`.  The `b:` variable returned the entire dict, but the `g:` variable returned the dict value (list).  Is this intentional?